File: Java/src/main/java/com/contidio/sdk/exceptions/OptimisticLockingBackendException.java
package com.contidio.sdk.exceptions;

import org.apache.commons.httpclient.HttpStatus;

import com.contidio.sdk.proto.MetadataProtocol.BackendErrorCode;

@SuppressWarnings("serial")
public class OptimisticLockingBackendException extends BackendException {
	public OptimisticLockingBackendException(long l, long m) {
		super(BackendErrorCode.OPTIMISTIC_LOCKING_FAILED, 
				"The optimistic locking versions are different (" + l + 
					" - " + m + ")", 
				"OptimisticLockingBackendException", 
				HttpStatus.SC_CONFLICT);
	}	
	
	public OptimisticLockingBackendException(String message) {
		super(BackendErrorCode.OPTIMISTIC_LOCKING_FAILED, message, "OptimisticLockingBackendException",
				HttpStatus.SC_CONFLICT);
	}	
}

File: .NET Core/Exceptions/OptimisticLockingBackendException.cs
using com.contidio.sdk.proto;
using System.Net;

namespace Contidio.Sdk.Exceptions
{
    public class OptimisticLockingBackendException : BackendException
    {
        public OptimisticLockingBackendException(long l, long m)
            : base(BackendErrorCode.OPTIMISTIC_LOCKING_FAILED,
				"The optimistic locking versions are different (" + l + 
					" - " + m + ")",
                "OptimisticLockingBackendException", 
                HttpStatusCode.Conflict)
        {
        }

        public OptimisticLockingBackendException(string message)
            : base(BackendErrorCode.OPTIMISTIC_LOCKING_FAILED, message, "OptimisticLockingBackendException", HttpStatusCode.Conflict)
        {
        }
    }
}
